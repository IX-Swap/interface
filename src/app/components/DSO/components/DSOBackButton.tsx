import React from 'react'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { Button } from '@material-ui/core'
import { IssuanceRoute } from 'app/pages/issuance/router/config'
import { useParams } from 'react-router-dom'

export const DSOBackButton = () => {
  const params = useParams()

  return (
    <Button
      component={AppRouterLinkComponent}
      to={IssuanceRoute.list} // TODO: add logic for the create DSO screen
      params={params}
    >
      Cancel
    </Button>
  )
}
