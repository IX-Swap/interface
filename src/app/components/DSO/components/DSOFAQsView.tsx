import React from 'react'
import { Grid, Typography } from '@mui/material'
import { DigitalSecurityOffering } from 'types/dso'
import { Expandable } from 'app/components/Expandable/Expandable'

export interface DSOFAQsViewProps {
  dso: DigitalSecurityOffering
}

export const DSOFAQsView = ({ dso }: DSOFAQsViewProps) => {
  return (
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <Typography variant='h5'>Most Asked Questions</Typography>
      </Grid>

      {dso.faqs.map((item, i) => {
        return (
          <Grid item xs={12} key={item.question}>
            <Expandable
              mainComponent={
                <Typography fontWeight={500} variant='subtitle1'>
                  {item.question}
                </Typography>
              }
              expandedComponent={
                <Typography
                  fontWeight={400}
                  color='textSecondary'
                  variant='body1'
                >
                  {item.answer}
                </Typography>
              }
              px={4}
              py={3}
              spacing={3}
              showArrow
            />
          </Grid>
        )
      })}
    </Grid>
  )
}
