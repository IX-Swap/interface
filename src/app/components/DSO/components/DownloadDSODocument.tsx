import React from 'react'
import { useDownloadRawFile } from 'hooks/useDownloadRawFile'
import { IconButton, Tooltip } from '@mui/material'
import { convertBlobToFile, openFileInNewTab } from 'hooks/utils'
import { Launch } from '@mui/icons-material'

export interface DownloadDSODocumentProps {
  dsoId: string
  documentId: string
  type?: 'document' | 'subscriptionDocument'
}

export const DownloadDSODocument = (props: DownloadDSODocumentProps) => {
  const { dsoId, documentId, type = 'document' } = props
  const uri =
    type === 'document'
      ? `/issuance/dso/dataroom/documents/raw/${dsoId}/${documentId}`
      : `/issuance/dso/dataroom/subscription/raw/${dsoId}`
  const [download, { isLoading }] = useDownloadRawFile(uri, {
    onSuccess: ({ data }) => {
      const file = convertBlobToFile(data, '')
      openFileInNewTab(file)
    }
  })
  const handleDownload = async () => await download()

  return (
    <Tooltip title='Download File'>
      <IconButton disabled={isLoading} onClick={handleDownload} size='large'>
        <Launch color='disabled' style={{ width: 23, height: 23 }} />
      </IconButton>
    </Tooltip>
  )
}
