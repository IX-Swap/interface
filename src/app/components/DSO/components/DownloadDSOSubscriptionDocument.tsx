import React from 'react'
import { useDownloadRawFile } from 'hooks/useDownloadRawFile'
import { Button, ButtonProps } from '@mui/material'
import { convertBlobToFile, openFileInNewTab } from 'hooks/utils'

export interface DownloadDSOSubscriptionDocumentProps extends ButtonProps {
  dsoId: string
}

export const DownloadDSOSubscriptionDocument = (
  props: DownloadDSOSubscriptionDocumentProps
) => {
  const { dsoId, children = 'Download', ...rest } = props
  const uri = `/issuance/dso/dataroom/subscription/raw/${dsoId}`
  const [download, { isLoading }] = useDownloadRawFile(uri, {
    onSuccess: ({ data }) => {
      const file = convertBlobToFile(data, '')
      openFileInNewTab(file)
    }
  })
  const handleDownload = async () => await download()

  return (
    <Button {...rest} onClick={handleDownload} disabled={isLoading}>
      {children}
    </Button>
  )
}
