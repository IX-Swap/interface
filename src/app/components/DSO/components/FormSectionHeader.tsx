import React from 'react'
import { Box, Typography } from '@material-ui/core'
import { useStyles } from 'app/components/DSO/components/FormSectionHeader.styles'
import { Variant } from '@material-ui/core/styles/createTypography'

export interface FormSectionHeaderProps {
  title: string
  variant?: Variant
}

export const FormSectionHeader = ({
  title,
  variant = 'h3'
}: FormSectionHeaderProps) => {
  const { formHeader } = useStyles()

  return (
    <Box className={formHeader} pb={0.5} mb={3}>
      <Typography variant={variant}>{title}</Typography>
    </Box>
  )
}
