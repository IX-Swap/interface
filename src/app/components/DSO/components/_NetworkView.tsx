import React from 'react'
import { useAllNetworks } from 'app/pages/accounts/pages/withdrawalAddresses/hooks/useAllNetworks'

export interface NetworkViewProps {
  networkId: string
}

export const NetworkView = (props: NetworkViewProps) => {
  const { networkId } = props
  const { data, isLoading } = useAllNetworks()
  const network = data?.find(network => network._id === networkId)

  return isLoading ? <>'...' </> : <>{network?.name}</>
}
