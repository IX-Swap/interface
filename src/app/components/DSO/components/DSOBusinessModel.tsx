import React from 'react'
import { TypedField } from 'components/form/TypedField'
import { RichTextEditor } from 'components/form/RichTextEditor'
import { wysiwygValueExtractor } from 'helpers/forms'
import { DSOContainer } from 'app/components/DSO/components/DSOContainer'
import { useFormContext } from 'react-hook-form'
import { DSOFormValues } from 'types/dso'

export const DSOBusinessModel = () => {
  const { control } = useFormContext<DSOFormValues>()

  return (
    <DSOContainer title='Business Model' item xs={12}>
      {/* @ts-ignore */}
      <TypedField
        control={control}
        component={RichTextEditor}
        customRenderer
        label='Business Model'
        name='businessModel'
        valueExtractor={wysiwygValueExtractor}
      />
    </DSOContainer>
  )
}
