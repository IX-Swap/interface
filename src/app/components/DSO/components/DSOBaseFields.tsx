import React from 'react'
import { Grid, TextField } from '@material-ui/core'
import { TypedField } from 'components/form/TypedField'
import { dateTimeValueExtractor, numericValueExtractor } from 'helpers/forms'
import { CorporateSelect } from 'components/form/CorporateSelect'
import { NetworkSelect } from 'components/form/NetworkSelect'
import { AssetSelect } from 'components/form/AssetSelect'
import { useFormContext } from 'react-hook-form'
import { DSOFormValues } from 'types/dso'
import { documentValueExtractor } from 'app/components/DSO/utils'
import { DataroomFileType } from 'config/dataroom'
import { DateTimePicker } from 'components/form/_DateTimePicker'
import { VSpacer } from 'components/VSpacer'
import { Dropzone } from 'components/dataroom/Dropzone'
import { CapitalStructureSelect } from 'components/form/CapitalStructureSelect'
import { NumericInput } from 'components/form/NumericInput'
import { positiveNumberFormat } from 'config/numberFormat'
import { FormSectionHeader } from 'app/components/DSO/components/FormSectionHeader'

export interface DSOBaseFieldsProps {
  isNew: boolean
  isLive: boolean
}

export const DSOBaseFields = (props: DSOBaseFieldsProps) => {
  const { isNew, isLive } = props
  const { control, watch } = useFormContext<DSOFormValues>()
  const capitalStructure = watch('capitalStructure')

  return (
    <Grid item>
      <VSpacer size='medium' />
      <Grid container direction='column' spacing={2}>
        <Grid item>
          <FormSectionHeader title='DSO Information' />
        </Grid>
        <Grid item>
          {/* @ts-ignore */}
          <TypedField
            customRenderer
            component={Dropzone}
            name='logo'
            label='Upload Logo'
            control={control}
            valueExtractor={documentValueExtractor}
            accept={DataroomFileType.image}
            documentInfo={{
              type: 'DSO Logo'
            }}
          />
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              <TypedField
                control={control}
                component={CapitalStructureSelect}
                label='Capital Structure'
                name='capitalStructure'
                helperText='Offering terms will be changed based on your capital structure'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <TypedField
                component={NetworkSelect}
                label='Network'
                name='network'
                disabled={!isNew}
                control={control}
                helperText='Select your blockchain network from the list'
                variant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              <TypedField
                component={TextField}
                label='Token Name'
                name='tokenName'
                disabled={isLive}
                control={control}
                helperText='Name of the token that describes your offering the best'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={2}>
              <TypedField
                component={TextField}
                label='Symbol'
                name='tokenSymbol'
                disabled={isLive}
                control={control}
                helperText='Token symbol'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={4}>
              <TypedField
                component={NumericInput}
                numberFormat={positiveNumberFormat}
                label='Decimal Places'
                name='decimalPlaces'
                disabled={capitalStructure?.toLowerCase() === 'equity'}
                control={control}
                helperText='Decimal Places'
                variant='outlined'
                valueExtractor={numericValueExtractor}
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={8}>
              <TypedField
                component={CorporateSelect}
                label='Corporate Structure'
                name='corporate'
                control={control}
                helperText='Select your corporate from the list'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={4}>
              <TypedField
                assetType='Currency'
                component={AssetSelect}
                label='Currency'
                name='currency'
                control={control}
                helperText='Your preferred currency'
                variant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              {/* @ts-ignore */}
              <TypedField
                component={DateTimePicker}
                customRenderer
                label='Launch Date'
                name='launchDate'
                control={control}
                disabled={isLive}
                valueExtractor={dateTimeValueExtractor}
                // @ts-expect-error
                defaultValue={null}
                helperText='Offering launch date'
                inputVariant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              {/* @ts-ignore */}
              <TypedField
                component={DateTimePicker}
                customRenderer
                label='Completion Date'
                name='completionDate'
                control={control}
                valueExtractor={dateTimeValueExtractor}
                // @ts-expect-error
                defaultValue={null}
                helperText='Offering completion date'
                inputVariant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}
