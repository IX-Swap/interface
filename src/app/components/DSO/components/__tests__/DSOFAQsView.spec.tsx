import React from 'react'
import { render, cleanup } from 'test-utils'
import { faqItem } from '__fixtures__/issuance'
import { DSOFAQsView } from 'app/components/DSO/components/DSOFAQsView'
import { DigitalSecurityOffering } from 'types/dso'
import { FormSectionHeader } from 'app/components/DSO/components/FormSectionHeader'
import * as Typography from '@material-ui/core'
import { dso } from '__fixtures__/authorizer'

jest.mock('@material-ui/core/Typography', () => jest.fn(() => null))

jest.mock('app/components/DSO/components/FormSectionHeader', () => ({
  FormSectionHeader: jest.fn(() => null)
}))

describe('DSOFAQsView', () => {
  const dsoObj: DigitalSecurityOffering = { ...dso, faqs: [faqItem] }
  afterEach(async () => {
    await cleanup()
    jest.clearAllMocks()
  })

  it('renders without error', () => {
    render(<DSOFAQsView dso={dsoObj} />)
  })

  it('renders FormSectionHeader', () => {
    render(<DSOFAQsView dso={dsoObj} />)

    expect(FormSectionHeader).toHaveBeenCalledTimes(0)
  })

  it('renders FormSectionHeader when isTitleVisible is true', () => {
    render(<DSOFAQsView dso={dsoObj} isTitleVisible />)

    expect(FormSectionHeader).toHaveBeenCalledTimes(1)
    expect(FormSectionHeader).toHaveBeenCalledWith(
      expect.objectContaining({ title: 'FAQs' }),
      {}
    )
  })

  it('renders content with correct props', () => {
    render(<DSOFAQsView dso={dsoObj} />)

    expect(Typography).toHaveBeenCalledTimes(2)

    expect(Typography).toHaveBeenNthCalledWith(
      1,
      expect.objectContaining({
        children: [1, '. ', faqItem.question],
        variant: 'subtitle1'
      }),
      {}
    )

    expect(Typography).toHaveBeenNthCalledWith(
      2,
      expect.objectContaining({
        children: faqItem.answer,
        variant: 'body1',
        style: { paddingLeft: 16, paddingTop: 16 }
      }),
      {}
    )
  })
})
