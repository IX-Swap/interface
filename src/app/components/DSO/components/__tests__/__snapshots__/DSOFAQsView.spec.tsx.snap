// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSOFAQsView matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-14eqi03-MuiTypography-root"
      >
        Most Asked Questions
      </h5>
    </div>
  </div>
</div>
`;
